### ISSUE TITLE
Pin python=2?

### ISSUE THREAD

**Issue by maddenp** [conda-forge/python-eccodes-feedstock#31]
I guess this is no surprise:
```
% conda create -y -n foo -c conda-forge -c defaults python-eccodes python=3
[ output elided ]
% source activate foo
(foo) % python -c 'import eccodes'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/pm/miniconda/envs/foo/lib/python3.5/site-packages/eccodes/__init__.py", line 1, in <module>
    from .eccodes import *
  File "/home/pm/miniconda/envs/foo/lib/python3.5/site-packages/eccodes/eccodes.py", line 1, in <module>
    from gribapi import __version__
  File "/home/pm/miniconda/envs/foo/lib/python3.5/site-packages/gribapi/__init__.py", line 1, in <module>
    from .gribapi import *                # noqa
  File "/home/pm/miniconda/envs/foo/lib/python3.5/site-packages/gribapi/gribapi.py", line 28, in <module>
    from gribapi import gribapi_swig as _internal
  File "/home/pm/miniconda/envs/foo/lib/python3.5/site-packages/gribapi/gribapi_swig.py", line 28, in <module>
    _gribapi_swig = swig_import_helper()
  File "/home/pm/miniconda/envs/foo/lib/python3.5/site-packages/gribapi/gribapi_swig.py", line 24, in swig_import_helper
    _mod = imp.load_module('_gribapi_swig', fp, pathname, description)
  File "/home/pm/miniconda/envs/foo/lib/python3.5/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "/home/pm/miniconda/envs/foo/lib/python3.5/imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: libhdf5.so.10: cannot open shared object file: No such file or directory
```
Would it be possible to pin `python=2` in `meta.yaml` to save users time trying to build Python 3 environments with `python-eccodes`?

**Comment by ocefpaf** [comment]
Version 0.13.0 was released with a Python 3 patch that works with an older hdf5. There is no need to pin anything b/c we don't build the new version on Python 3.

@conda-forge/python-eccodes maybe we should consider removing the older `0.13.0` builds.

**Comment by maddenp** [comment]
Shouldn't conda's resolver have been able to reject the above list of packages, since we know that `python-eccodes` isn't compatible with `python=3`? Maybe I'm missing something... I'm surprised that I was allowed to create this environment only to have it immediately fail. Couldn't that be prevented with pinning? Apologies if I'm missing something...

**Comment by maddenp** [comment]
I wonder if I misunderstood: Maybe you weren't saying there is no problem but, rather, were suggesting a superior (to pinning) solution.

**Comment by ocefpaf** [comment]
> I wonder if I misunderstood

Yep. What I said was: Version 0.13.0 of python-eccodes was released for Python 3 using a patch for that. However, that versions is too old and incompatible with the modern stack of dependencies on conda-forge. There is nothing wrong with the newer packages nor with the Python 2 dependency.

You probably got the old version, 0.13.0, when you tried to install it with Python 3.

BTW I removed the 0.13.0 version so that problem should not happen again.
